### PR TITLE
add mysql builtin function cos,from_base64,tan,cot

### PR DESCRIPTION
**Pull request by qiukeren**
add mysql builtin function COS and test case, 
and test success on sierra with go 1.8.

### CODE REVIEW

**Comment by shenli** [comment]
@qiukeren Thanks for your PR! Please sign the CLA.

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -501,6 +501,33 @@ func (s *testEvaluatorSuite) TestSin(c *C) {
...
 
+func (s *testEvaluatorSuite) TestCos(c *C) {
+	defer testleak.AfterTest(c)()
+	tbl := []struct {
+		Arg interface{}
+		Ret interface{}
+	}{
+		{nil, nil},
+		{int64(0), float64(1)},
+		{math.Pi, float64(-1)}, // Pie ==> 0
```
Comment (shenli) [inline expression/builtin_math_test.go]: The comment is not correct.

**Comment by qiukeren** [comment]
The merge with qiukeren/master comes with no change.

**Comment by shenli** [comment]
LGTM
@coocood @XuHuaiyu @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM
Thanks for your contribution! @qiukeren

**Pull request merged**